### DISCUSSION TITLE
Codegen Compatibility?

### DISCUSSION THREAD

**Discussion by smgonzales** [mirumee/ariadne#1146] [General]
Hi, Ariadne docs indicate that it is compatible with Python 3.7 or higher.  Is that true of codegen as well or does it require higher versions of Python?  

It’s also not compatible with a graphql-core lower than 3.x.  Is that correct?

Thanks!!!

**Comment by rafalp** [comment]
Hello,

Ariadne is developed against graphql-core 3.2.

Ariadne Codegen requires Python 3.9 or higher because of `ast.unparse` function not present in previous versions. But clients generated by it should work with Python 3.7.

**Reply by smgonzales** [reply]
Thanks for the information!
